### PR TITLE
Feedback API

### LINKED ISSUE #2350: Custom feedback api
See https://github.com/toggl/backend/issues/6310 for technical details (WIP at point of writing).

The model will probably look something like:

```
class X
{
    string Email
    string Message
    Dictionary<string, string>/IEnumerable<KeyValuePair<string, string>> Data
}
```

This should serialise trivially to the needed format.

### PR DESCRIPTION
**Pull request by simonrozsival**
Closes #2350 

~On-hold: https://github.com/toggl/toggl_api/pull/828 must be finished and deployed to staging before we can start testing this +~ it must be deployed into production before we can merge this into `develop` (so it can be part of the next release for any platform).

WIP: we still don't know how exactly will the API work and if this will work as is or if it will need any further changes + also whether the integration tests are sufficient or if we need any more (or less).

Integration tests: https://app.bitrise.io/build/315c7c2024950136

### CODE REVIEW

File: Toggl.Ultrawave/ApiClients/Interfaces/IFeedbackApiClient.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive;
+using Toggl.Multivac;
+
+namespace Toggl.Ultrawave.ApiClients
+{
+    public interface IFeedbackApiClient
```
Comment (juanlaube) [inline Toggl.Ultrawave/ApiClients/Interfaces/IFeedbackApiClient.cs]: Why not `IFeedbackApi` to be consistent with other `WhateverApi`s we have?

**Comment by simonrozsival** [comment]
The API PR is still not merged.

**Comment by daividssilverio** [comment]
Integration tests: https://app.bitrise.io/build/47068c04340fb2ee
The Feedback api related integration tests are passing. 
There is something going on with the integration tests for the workspaces/id/plans. I've asked backed about it (we are getting 502 bad gateway)

File: Toggl.Ultrawave.Tests/ApiClients/FeedbackApiTests.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            public async Task SerializesTheJsonCorrectly()
+            {
+                var email = Email.From($"{Guid.NewGuid()}@toggl.space");
+                var message = "XYZ.";
+                var data = new Dictionary<string, string>
+                {
+                    ["device"] = "SomePhone",
+                    ["some random key"] = "some also random value"
+                };
+                var serializedJson = $"{{\"email\":\"{email}\",\"message\":\"{message}\",\"data\":[{{\"key\":\"device\",\"value\":\"SomePhone\"}},{{\"key\":\"some random key\",\"value\":\"some also random value\"}}]}}";
```
Comment (juanlaube) [inline Toggl.Ultrawave.Tests/ApiClients/FeedbackApiTests.cs]: You're repeating the values in `data` here. Use `data["device"]` and `data["some random key"]`.

**Comment by daividssilverio** [comment]
I'm merging this with partially successful integration tests. Related feedback api integration tests are passing.

**Pull request merged**